Userland/Applications/Browser/BrowserWindow.cpp

```diff
@@ -262,31 +263,40 @@ void BrowserWindow::build_menus()
     m_search_engine_actions.add_action(*m_disable_search_engine_action);
     m_disable_search_engine_action->set_checked(true);
 
-    auto search_engines_file = Core::File::construct(Browser::search_engines_file_path());
-    if (search_engines_file->open(Core::OpenMode::ReadOnly)) {
-        if (auto maybe_json = JsonValue::from_string(search_engines_file->read_all()); !maybe_json.is_error() && maybe_json.value().is_array()) {
-            auto json = maybe_json.release_value().as_array();
-            for (auto& json_item : json.values()) {
-                if (!json_item.is_object())
-                    continue;
-                auto search_engine = json_item.as_object();
-                auto name = search_engine.get("title").to_string();
-                auto url_format = search_engine.get("url_format").to_string();
-
-                auto action = GUI::Action::create_checkable(
-                    name, [&, url_format](auto&) {
-                        g_search_engine = url_format;
-                        Config::write_string("Browser", "Preferences", "SearchEngine", g_search_engine);
-                    },
-                    this);
-                search_engine_menu.add_action(action);
-                m_search_engine_actions.add_action(action);
-
-                if (g_search_engine == url_format) {
-                    action->set_checked(true);
-                    search_engine_set = true;
+    auto maybe_search_engines_file = Core::Stream::File::open(Browser::search_engines_file_path(), Core::Stream::OpenMode::Read);
```

Comment:
This gets indented quite far to the right and any error state is not handled explicitly.
Perhaps it would be more readable if you put it in a lambda which you immediately call.
Something like:
```cpp

[&] {
    auto code_here;
    if (maybe_search_engines_file.is_error())
        return;
     
     if (some_other_error())
        return;
    
     auto more_code;

    auto json = maybe_json.release_value().as_array();
    for (auto& json_item : json.values()) {
        if (!json_item.is_object())
            continue;
        auto search_engine = json_item.as_object();
        // etc.
    }
}();
```

Although I do realize that it may look like you return from the outer function...

Reply:
Thanks. Yeah I've actually decided to wrap the search-engines code into a method that returns ErrorOr so I can TRY inside it, it's a lot better than what I had.

Reply:
Of course, that's even better.